python/semantic_kernel/connectors/search/bing/bing_search.py

```diff
@@ -0,0 +1,216 @@
+            custom_config: The Bing Custom Search instance's unique identifier.
+                If provided, will override the value in the env vars or .env file.
+            client: Provide a client to use for the Bing Search API.
+            env_file_path: The optional path to the .env file. If provided,
+                the settings are read from this file path location.
+            env_file_encoding: The optional encoding of the .env file.
+        """
+        try:
+            settings = BingSettings.create(
+                api_key=api_key,
+                custom_config=custom_config,
+                env_file_path=env_file_path,
+                env_file_encoding=env_file_encoding,
+            )
+        except ValidationError as ex:
+            raise ServiceInitializationError("Failed to create Bing settings.") from ex
+
+        super().__init__(settings=settings)
+
+    async def search(
+        self, query: str, options: TextSearchOptions | None = None, **kwargs: Any
+    ) -> "KernelSearchResults[str]":
+        """Search for text, returning a KernelSearchResult with a list of strings."""
+        options = self._get_options(options, **kwargs)
+        results = await self._inner_search(query=query, options=options)
+        return KernelSearchResults(
+            results=self._get_result_strings(results),
+            total_count=self._get_total_count(results, options),
+            metadata=self._get_metadata(results),
+        )
+
+    async def get_text_search_results(
+        self, query: str, options: TextSearchOptions | None = None, **kwargs
+    ) -> "KernelSearchResults[TextSearchResult]":
+        """Search for text, returning a KernelSearchResult with TextSearchResults."""
+        options = self._get_options(options, **kwargs)
+        results = await self._inner_search(query=query, options=options)
+        return KernelSearchResults(
+            results=self._get_text_search_results(results),
+            total_count=self._get_total_count(results, options),
+            metadata=self._get_metadata(results),
+        )
+
+    async def get_search_results(
+        self, query: str, options: TextSearchOptions | None = None, **kwargs
+    ) -> "KernelSearchResults[BingWebPage]":
+        """Search for text, returning a KernelSearchResult with the results directly from the service."""
+        options = self._get_options(options, **kwargs)
+        results = await self._inner_search(query=query, options=options)
+        return KernelSearchResults(
+            results=self._get_bing_web_pages(results),
+            total_count=self._get_total_count(results, options),
+            metadata=self._get_metadata(results),
+        )
+
+    async def _get_result_strings(self, response: BingSearchResponse) -> AsyncIterable[str]:
+        if response.web_pages is None:
+            return
+        for web_page in response.web_pages.value:
+            yield web_page.snippet or ""
```

Comment:
Question: what is the advantage of yielding over returning a list?

Reply:
This is to unify the interface across services that can do things like paging and it therefore might not have to load everything first if the app only uses the first few pieces. And also in sync with dotnet